### PR TITLE
Add more tests for PathNormalizer

### PR DESCRIPTION
**Pull request by aeons**
I'm not convinced the property tests makes sense.

### CODE REVIEW

File: server/src/test/scala/org/http4s/server/staticcontent/PathNormalizerSpec.scala [hunk 1]
```diff
@@ -1,18 +1,69 @@
...
+        "/a/b/c/g/./h" ! "/a/b/c/g/h" |
+        "/a/b/c/g/../h" ! "/a/b/c/h" |
+        "/a/b/c/g;x=1/./y" ! "/a/b/c/g;x=1/y" |
+        "/a/b/c/g;x=1/../y" ! "/a/b/c/y" | { (path, expected) =>
+        PathNormalizer.removeDotSegments(path) must_=== expected
+      }
+    }
 
-      PathNormalizer.removeDotSegments(path1) must_=== "/a/g"
+    "remove dot segments correctly in other examples" >> prop { input: String =>
```
Comment (rossabaker) [inline server/src/test/scala/org/http4s/server/staticcontent/PathNormalizerSpec.scala]: This is rarely going to generate `..` and `.`, or even multiple segments.  I think for this to make sense, we need to have a smarter generator.

And if we do have a property test, ensuring there are no more `./` or `../` would be good assertions.

Comment (aeons) [reply server/src/test/scala/org/http4s/server/staticcontent/PathNormalizerSpec.scala]: The prop is called with `setGen`, with a custom generator. It's not perfect, but it does create multiple segments and dots.

**Comment by aeons** [comment]
What's going on here: https://travis-ci.org/http4s/http4s/jobs/287183028#L773

**Comment by aeons** [comment]
I rebased on top of e7043e9fc97dde6efe9006a8ddcdb6c668de0dbc, which is only in the merge to 0.17 and is probably why the build broke.

File: server/src/test/scala/org/http4s/server/staticcontent/PathNormalizerSpec.scala (hunk 1 above)
Comment (rossabaker) [reply server/src/test/scala/org/http4s/server/staticcontent/PathNormalizerSpec.scala]: Oh, I missed that the first time. 😬

**Pull request merged**